tiledb/api/src/group/mod.rs

```diff
@@ -0,0 +1,745 @@
+    }
+
+    pub fn member(&self, key: LookupKey) -> TileDBResult<GroupInfo> {
+        let c_context = self.context.capi();
+        let mut tiledb_uri: *mut ffi::tiledb_string_t = out_ptr!();
+        let mut tiledb_type: ffi::tiledb_object_t = out_ptr!();
+        let name: String = match key {
+            LookupKey::Index(index) => {
+                let mut tiledb_name: *mut ffi::tiledb_string_t = out_ptr!();
+                self.context.capi_return(unsafe {
+                    ffi::tiledb_group_get_member_by_index_v2(
+                        c_context,
+                        Self::capi(self),
+                        index.try_into().unwrap(),
+                        &mut tiledb_uri as *mut *mut ffi::tiledb_string_t,
+                        &mut tiledb_type,
+                        &mut tiledb_name as *mut *mut ffi::tiledb_string_t,
+                    )
+                })?;
+                let name = TDBString {
+                    raw: RawTDBString::Owned(tiledb_name),
+                }
+                .to_string();
+                let name = name?;
+                Ok(name) as TileDBResult<String>
+            }
+            LookupKey::Name(name) => {
+                let c_name = cstring!(name.as_ref() as &str);
+                self.context.capi_return(unsafe {
+                    ffi::tiledb_group_get_member_by_name_v2(
+                        c_context,
+                        Self::capi(self),
+                        c_name.as_ptr(),
+                        &mut tiledb_uri as *mut *mut ffi::tiledb_string_t,
+                        &mut tiledb_type,
+                    )
+                })?;
+                Ok(name.to_owned())
+            }
+        }?;
+
+        let uri = TDBString {
+            raw: RawTDBString::Owned(tiledb_uri),
+        }
+        .to_string();
+        let uri = uri?;
+
+        let object_type = ObjectType::try_from(tiledb_type)?;
+        Ok(GroupInfo {
+            uri,
+            group_type: object_type,
+            name,
+        })
+    }
+
+    pub fn is_relative_uri<S>(&self, name: S) -> TileDBResult<bool>
+    where
+        S: AsRef<str>,
+    {
+        let c_context = self.context.capi();
+        let mut c_relative: u8 = out_ptr!();
+        let c_name = cstring!(name.as_ref());
+        self.context.capi_return(unsafe {
+            ffi::tiledb_group_get_is_relative_uri_by_name(
+                c_context,
+                Self::capi(self),
+                c_name.as_ptr(),
+                &mut c_relative,
+            )
+        })?;
+        Ok(c_relative > 0)
+    }
+
+    pub fn is_open(&self) -> TileDBResult<bool> {
+        let c_context = self.context.capi();
+        let mut c_open: i32 = out_ptr!();
+        self.context.capi_return(unsafe {
+            ffi::tiledb_group_is_open(c_context, Self::capi(self), &mut c_open)
+        })?;
+        Ok(c_open > 0)
+    }
+
+    pub fn dump(&self, recursive: bool) -> TileDBResult<Option<String>> {
```

Comment:
Does this ever return `None`?

Reply:
Yes, when the dump is empty.

Reply:
Which LOC does that? I don't see anything returning `None`

Reply:
Fixed.